gatsby/copyingplonesite.rst

```diff
@@ -0,0 +1,251 @@
+Copying The Plone Site
+======================
+
+Now we have seen how the plugin and starter work together.
+
+The end goal of the plugin along with GatsbyJS is to generate a static site which is an exact copy of the Plone site it sourced data from.
+
+Once ``gatsby-source-plugin`` retrieves all the required data for us, the ``gatsby-starter-plone`` processes and uses this data to generate the static site.
+
+Internally what ``gatsby-starter-plone`` does in steps is:
+
+- Create pages for each content object and ensure tree structure by retaining parent and items relationships.
+- Display HTML, images and files in content objects correctly.
+- Handle navigation in the site with a Navbar and Breadcrumbs.
+
+
+Page Creation
+-------------
+
+``exports.createPages`` is the API used in ``gatsby-node.js`` for creating pages from nodes.
+
+To create pages for all nodes, first we query the list of all different types of nodes and use this list to create individual pages.
+
+To illustrate how this process works, let us create pages for all nodes of type ``PloneFolder``.
+
+Later we can move on to include all the other types.
+
+In ``gatsby-node.js``:
+
+.. code-block:: javascript
+
+  const path = require('path');
+
+  exports.createPages = async ({ graphql, actions }) => {
+    const { createPage } = actions;
+
+    // Get data via GraphQL
+    const result = await graphql(`
+      {
+        allPloneFolder {
+          edges {
+            node {
+              _path
+            }
+          }
+        }
+      }
+    `);
+
+    // Create pages for each PloneFolder item
+    result.data.allPloneFolder.edges.forEach(({ node }) => {
+      createPage({
+        path: node._path,
+        component: path.resolve('./src/templates/Folder.js'),
+      });
+    });
+  };
+
+The ``_path`` property is very helpful to create pages.
+It is unique to all nodes.
+Files and images are linked to the nodes they are present in, with the ``_path`` value.
+
+This backlinking will be explained in the later sections.
+
+So basically it can be used to set the relative path of a node in the static site generated.
+
+It can also get the required images and files.
```

Comment:
Combine 2 paragraphs into 1, and minor edit:

```
Backlinking can be used to set the relative path of a node in the generated static site.
It can also get the required images and files.
```